ui/pages/confirm-approve/confirm-approve-content/confirm-approve-content.component.js

```diff
@@ -85,15 +87,43 @@ export default class ConfirmApproveContent extends Component {
     footer,
     noBorder,
   }) {
-    const { supportsEIP1559 } = this.props;
+    const { supportsEIP1559, txData } = this.props;
+    const { userAcknowledgedGasMissing } = this.state;
     const { t } = this.context;
+    const hasSimulationError = Boolean(txData.simulationFails);
+    const renderSimulationFailureWarning =
+      hasSimulationError && !userAcknowledgedGasMissing;
     return (
       <div
         className={classnames({
           'confirm-approve-content__card': !noBorder,
           'confirm-approve-content__card--no-border': noBorder,
         })}
       >
+        {renderSimulationFailureWarning && (
+          <Box
+            paddingTop={0}
+            paddingRight={4}
+            paddingBottom={4}
+            paddingLeft={4}
+          >
+            <ActionableMessage
+              message={t('simulationErrorMessageV2')}
+              useIcon
+              iconFillColor="var(--color-error-default)"
+              type="danger"
+              primaryActionV2={
+                userAcknowledgedGasMissing === true
+                  ? undefined
+                  : {
+                      label: t('proceedWithTransaction'),
+                      onClick: () =>
+                        this.setState({ userAcknowledgedGasMissing: true }),
+                    }
+              }
+            />
+          </Box>
+        )}
```

Comment:
Might be useful to extract out a small component for above warning display. It seems to be repeated a few places.